extras/kinesis-asl/src/main/scala/org/apache/spark/streaming/kinesis/KinesisTestUtils.scala

```diff
@@ -264,3 +237,33 @@ private[kinesis] object KinesisTestUtils {
     }
   }
 }
+
+/** A wrapper interface that will allow us to consolidate the code for synthetic data generation. */
+private[kinesis] trait KinesisProducer {
```

Comment:
Lets name it `KinesisDataGenerator`, to avoid confusion with "producer" in KPL. 
KinesisProducer --> KinesisDataGenerator
KinesisClientProducer --> SimpleDataGenerator
KinesisProducerLibraryProducer --> KPLDataGenerator